mlflow/tracking/fluent.py

```diff
@@ -714,6 +715,19 @@ def flush_artifact_async_logging() -> None:
         _artifact_repo.flush_async_logging()
 
 
+def flush_trace_async_logging(terminate=False) -> None:
```

Comment:
Q: We currently have separate function for flushing run data and artifacts, so I just followed the convention and added a new one for tracing. However, it might be simpler for users to have a single `flush_async_logging()` function and use parameters if granular control is required.

Reply:
Building an additional higher-level single purpose thread pool flush API seems like a good idea. I think we'll still want the logic separated for each one as it is (maintaining a public state for each), but having the higher-level multi-purpose simplified API be included in docs / examples.

Reply:
Makes sense, I will file a follow-up for adding a new unified API while keeping the original ones. In fact, it seems we don't have a doc for async logging so will try adding it in the same PR too🙂